feature-libs/product-multi-dimensional/list/root/components/price-range/product-multi-dimensional-price-range.component.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { EMPTY, Observable } from 'rxjs';
+import { Product } from '@spartacus/core';
+import { ProductListItemContext } from '@spartacus/storefront';
+
+@Component({
+  selector: 'cx-product-multi-dimensional-price-range',
+  templateUrl: './product-multi-dimensional-price-range.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProductMultiDimensionalPriceRangeComponent {
+  productListItemContext?: ProductListItemContext = inject(
+    ProductListItemContext
+  );
+  readonly product$: Observable<Product> =
+    this.productListItemContext?.product$ ?? EMPTY;
+
+  getProductPrice(product: Product) {
+    const priceRange = product.priceRange;
+    const isMultiDimensionalAndHasPriceRange =
+      product.multidimensional && Object.values(priceRange).length;
+
+    if (isMultiDimensionalAndHasPriceRange) {
+      const maxPrice = priceRange.maxPrice.formattedValue;
+      const minPrice = priceRange.minPrice.formattedValue;
+      return `${minPrice} - ${maxPrice}`;
+    }
+
+    return product.price?.formattedValue ?? 0;
+  }
```

Comment:
```suggestion
  getProductPrice(product: Product) {
    const defaultValue = '0';

    if (!product.multidimensional) {
      return product.price?.formattedValue ?? '0';
    }

    const priceRange = product.priceRange;
    if (!priceRange || !priceRange.maxPrice || !priceRange.minPrice) {
      return product.price?.formattedValue ?? '0';
    }

    const maxPrice = priceRange.maxPrice.formattedValue;
    const minPrice = priceRange.minPrice.formattedValue;
    return `${minPrice} - ${maxPrice}`;
  }
```
Due to TS issues and few places related to the `priceRange`, I propopose vew improvements:

- Added early return if product is not `multidimensional`.
- Added early return if `priceRange`, `priceRange.maxPrice`, or `priceRange.minPrice` is not defined.
- Moved `0` to a separate constant and returned it as a default value, ensuring a consistent return type.

Reply:
fixed

Reply:
on second thought, this is wrong. This is what I propose:
```typescript
  getProductPrice(product: Product): string {
    const defaultValue = '0';
    if (!product.multidimensional) {
      return product.price?.formattedValue ?? defaultValue;
    }
    const priceRange = product.priceRange;

    const maxPrice = priceRange?.maxPrice?.formattedValue;
    const minPrice = priceRange?.minPrice?.formattedValue;

    return maxPrice && minPrice ? `${minPrice} - ${maxPrice}` : '';
  }
``` 
if there is no price range it should not return any price, as it would be misleading